testdata/project-v2/go.mod

```diff
@@ -6,6 +6,7 @@ require (
 	github.com/go-logr/logr v0.1.0
 	github.com/onsi/ginkgo v1.12.1
 	github.com/onsi/gomega v1.10.1
+	golang.org/x/sys v0.0.0-20220330033206-e17cdc41300f // indirect; we enforce this version to supports go 1.18+
```

Comment:
We need to check this one.
Should we not update go/v2 to also use go 1.18 or not?

Reply:
TBH, I don't see any reason not to upgrade go/v2 as well. This is the only change we introducing to go/v2 other than specifying the version

Reply:
As raised and checked by @varshaprasad96 we should not do that.
Why? it might introduce issues to the users since the go version will not be compatible with the deps used.